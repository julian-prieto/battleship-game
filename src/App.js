import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Navigation } from "components";
import { Home, Game, Score } from "views";

import styles from "./styles.module.scss";

export default function App() {
  return (
    <Router>
      <div className={styles.main}>
        <div className={styles.container}>
          <Navigation />
          <Switch>
            <Route path="/game">
              <Game />
            </Route>
            <Route path="/scores">
              <Score />
            </Route>
            <Route path="/">
              <Home />
            </Route>
          </Switch>
        </div>
      </div>
    </Router>
  );
}
