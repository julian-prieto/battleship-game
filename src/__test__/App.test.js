import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import App from "../App";

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Matches Snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
