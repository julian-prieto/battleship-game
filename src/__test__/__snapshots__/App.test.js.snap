// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  className="main"
>
  <div
    className="container"
  >
    <nav>
      <ul
        className="ul"
        data-testid="sections-list"
      >
        <li
          className="li"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="li"
        >
          <a
            href="/game"
            onClick={[Function]}
          >
            Play
          </a>
        </li>
        <li
          className="li"
        >
          <a
            href="/scores"
            onClick={[Function]}
          >
            My Scores
          </a>
        </li>
      </ul>
    </nav>
    <main
      className="main"
    >
      <span
        className="title"
        data-testid="welcome"
      >
        Welcome to Battleship Game
      </span>
      <img
        alt="Splash"
        src="splash.png"
        width={450}
      />
    </main>
  </div>
</div>
`;
